packages/svelte/src/lib/chat.svelte.ts

```diff
@@ -0,0 +1,341 @@
+} from "@ai-sdk/ui-utils";
+import { isAbortError } from "@ai-sdk/provider-utils";
+import {
+  KeyedChatStore,
+  getChatContext,
+  hasChatContext,
+} from "./chat-context.svelte.js";
+
+export type ChatOptions = Readonly<
+  Omit<UseChatOptions, "keepLastMessageOnError"> & {
+    /**
+     * Maximum number of sequential LLM calls (steps), e.g. when you use tool calls.
+     * Must be at least 1.
+     * A maximum number is required to prevent infinite loops in the case of misconfigured tools.
+     * By default, it's set to 1, which means that only a single LLM call is made.
+     * @default 1
+     */
+    maxSteps?: number;
+  }
+>;
+
+export type { CreateMessage, Message, UIMessage };
+
+export class Chat {
+  readonly #options: ChatOptions = {};
+  readonly #api = $derived(this.#options.api ?? "/api/chat");
+  readonly #generateId = $derived(this.#options.generateId ?? generateId);
+  readonly #id = $derived(this.#options.id ?? this.#generateId());
+  readonly #maxSteps = $derived(this.#options.maxSteps ?? 1);
+  readonly #streamProtocol = $derived(this.#options.streamProtocol ?? "data");
+  readonly #keyedStore = $state<KeyedChatStore>()!;
+  readonly #store = $derived(this.#keyedStore.get(this.#id));
+  #abortController: AbortController | undefined;
+
+  /**
+   * The id of the chat. If not provided through the constructor, a random ID will be generated
+   * using the provided `generateId` function, or a built-in function if not provided.
+   */
+  get id() {
+    return this.#id;
+  }
+
+  /**
+   * Additional data added on the server via StreamData.
+   *
+   * This is writable, so you can use it to transform or clear the chat data.
+   */
+  get data() {
+    return this.#store.data;
+  }
+  set data(value: JSONValue[] | undefined) {
+    this.#store.data = value;
+  }
+
+  /**
+   * Hook status:
+   *
+   * - `submitted`: The message has been sent to the API and we're awaiting the start of the response stream.
+   * - `streaming`: The response is actively streaming in from the API, receiving chunks of data.
+   * - `ready`: The full response has been received and processed; a new user message can be submitted.
+   * - `error`: An error occurred during the API request, preventing successful completion.
+   */
+  get status() {
+    return this.#store.status;
+  }
+
+  /** The error object of the API request */
+  get error() {
+    return this.#store.error;
+  }
+
+  /** The current value of the input. Writable, so it can be bound to form inputs. */
+  input = $state<string>()!;
+
+  /**
+   * Current messages in the chat.
+   *
+   * This is writable, which is useful when you want to edit the messages on the client, and then
+   * trigger {@link reload} to regenerate the AI response.
+   */
+  get messages(): UIMessage[] {
+    return this.#store.messages;
+  }
+  set messages(value: Message[]) {
+    this.#store.messages = fillMessageParts(value);
+  }
+
+  constructor(options: ChatOptions = {}) {
+    if (hasChatContext()) {
```

Comment:
This block means that, for the vast majority of use cases, users don't need to call `createAIContext`, because these class instances can just manage their own state. If the users _need_ the shared-between-instances state, they can call that function in one of their layouts.